src/main/scala/is/hail/expr/ir/functions/ArrayFunctions.scala

```diff
@@ -0,0 +1,131 @@
+package is.hail.expr.ir.functions
+
+import is.hail.expr.ir._
+import is.hail.expr.types._
+import is.hail.expr.types.coerce
+
+object ArrayFunctions extends RegistryFunctions {
+
+  def registerAll() {
+    registerIR("size", TArray(tv("T")))(ArrayLen)
+
+    val arrayOps =
```

Comment:
@jigold this (and all the functions registered in the following `for` loop) is the only actual addition to this file; everything else was moved verbatim from UtilFunctions.

Reply:
(likewise, everything that got deleted from `UtilFunctions` was deleted because it got moved here.)